src/ngraph/partial_shape.cpp

```diff
@@ -36,6 +36,65 @@ bool ngraph::PartialShape::is_static() const
                                            [](const Dimension& d) { return d.is_static(); });
 }
 
+bool ngraph::PartialShape::operator==(const PartialShape& partial_shape) const
+{
+    if (rank() != partial_shape.rank())
+    {
+        return false;
+    }
+    if (rank().is_dynamic())
```

Comment:
Is this "If" condition correct? I think you are trying to say "If one of the two are dynamic".
The "If" is to only allow "the two are both static with same rank or both dynamic" to fall to this condition.
The "If" below assumes "both are static rank".

Reply:
The if at 41 ensures both ranks are equal, so the one in 45 implies both are dynamic.